internal/cmd/beta/cdn/distribution/list/list.go

```diff
@@ -0,0 +1,173 @@
+				`$ stackit beta dns distribution list --sort-by=id`,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background() // should this be cancellable?
+
+			model, err := parseInput(params.Printer, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			apiClient, err := client.ConfigureClient(params.Printer, params.CliVersion)
+			if err != nil {
+				return err
+			}
+
+			distributions, err := fetchDistributions(ctx, model, apiClient)
+			if err != nil {
+				return fmt.Errorf("fetch distributions: %w", err)
+			}
+
+			return outputResult(params.Printer, model.OutputFormat, distributions)
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+var sortByFlagOptions = []string{"id", "created", "updated", "origin-url", "status"}
+
+func configureFlags(cmd *cobra.Command) {
+	// same default as apiClient
+	cmd.Flags().Var(flags.EnumFlag(false, "created", sortByFlagOptions...), sortByFlag, fmt.Sprintf("Sort entries by a specific field, one of %q", sortByFlagOptions))
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, _ []string) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		SortBy:          flags.FlagWithDefaultToStringValue(p, cmd, sortByFlag),
+	}
+
+	p.DebugInputModel(model)
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *cdn.APIClient, nextPageID cdn.ListDistributionsResponseGetNextPageIdentifierAttributeType) cdn.ApiListDistributionsRequest {
+	req := apiClient.ListDistributions(ctx, model.GlobalFlagModel.ProjectId)
+	req = req.SortBy(toAPISortBy(model.SortBy))
+	req = req.PageSize(100)
+	if nextPageID != nil {
+		req = req.PageIdentifier(*nextPageID)
+	}
+	return req
+}
+
+func toAPISortBy(sortBy string) string {
+	switch sortBy {
+	case "id":
+		return "id"
+	case "created":
+		return "createdAt"
+	case "updated":
+		return "updatedAt"
+	case "origin-url":
+		return "originUrl"
+	case "status":
+		return "status"
+	default:
+		panic("invalid sortBy value, programmer error")
+	}
+}
+
+func outputResult(p *print.Printer, outputFormat string, distributions []cdn.Distribution) error {
+	if distributions == nil {
+		distributions = make([]cdn.Distribution, 0) // otherwise prints null in json output
+	}
+	return p.OutputResult(outputFormat, distributions, func() error {
+		if len(distributions) == 0 {
+			p.Outputln("No CDN distributions found")
+			return nil
+		}
+
+		table := tables.NewTable()
+		table.SetHeader("ID", "REGIONS", "STATUS")
+		for i := range distributions {
+			d := &distributions[i]
+			regions := make([]string, 0, len(*d.Config.Regions))
+			for _, r := range *d.Config.Regions {
+				regions = append(regions, string(r))
+			}
+			joinedRegions := strings.Join(regions, ", ")
+			table.AddRow(
+				utils.PtrString(d.Id),
+				joinedRegions,
```

Comment:
```suggestion
				strings.Join(sdkUtils.EnumSliceToStringSlice(*d.Config.Regions), ","),
```

try this one please :smile: 

https://github.com/stackitcloud/stackit-sdk-go/blob/0437d95e48ffe7fea9ffe08f152229ff401419e3/core/utils/utils.go#L17-L29